### PR TITLE
Implement `create_block` method

### PR DESCRIPTION
**Pull request by TomTaehoonKim**

### CODE REVIEW

**Comment by TomTaehoonKim** [comment]
CI failing due to dependency to #180. Will be fine after merge and rebase.

File: repository/src/lib.rs [hunk 1]
```diff
@@ -424,9 +425,112 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+            repository_merkle_root: Hash256::zero(), // TODO
+            validator_set: reserved_state.create_validator_set().unwrap(),
+            version: "0.0.0".to_string(),
+        };
+        let block_commit = Commit::Block(block_header.clone());
+        let semantic_commit = to_semantic_commit(&block_commit);
+
+        self.raw.checkout_clean().await?;
+        self.raw.checkout(WORK_BRANCH_NAME.into()).await?;
+        let result = self.raw.create_semantic_commit(semantic_commit).await?;
```
Comment (YoonhoKim0527) [inline repository/src/lib.rs:520]: How about result -> commit_hash? It seems more acceptable.
```suggestion
        let commit_hash = self.raw.create_semantic_commit(semantic_commit).await?;
```

File: repository/src/lib.rs [hunk 2]
```diff
@@ -424,9 +425,112 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+                    ExtraAgendaTransaction::Delegate(tx) => {
+                        reserved_state
+                            .apply_delegate(&tx)
+                            .map_err(|e| anyhow!("invalid delegation: {}", e))?;
+                    }
+                    ExtraAgendaTransaction::Undelegate(tx) => {
+                        reserved_state
+                            .apply_undelegate(&tx)
+                            .map_err(|e| anyhow!("invalid undelegation: {}", e))?;
+                    }
```
Comment (junha1) [inline repository/src/lib.rs:478]: This must have been already checked by CSV.

File: repository/src/lib.rs [hunk 3]
```diff
@@ -424,9 +425,112 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+                }
+            }
+        }
+
+        // Verify `finalization_proof`
+        let fp_commit_hash = self.raw.locate_branch(FP_BRANCH_NAME.into()).await?;
+        let fp_semantic_commit = self.raw.read_semantic_commit(fp_commit_hash).await?;
+        let finalization_proof = fp_from_semantic_commit(fp_semantic_commit).unwrap().proof;
+        verify::verify_finalization_proof(&last_header, &finalization_proof)
+            .map_err(|e| anyhow!(e))?;
```
Comment (junha1) [inline repository/src/lib.rs:496]: Why do we check this? This is already guaranteed.

File: repository/src/lib.rs [hunk 4]
```diff
@@ -424,9 +425,112 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+
+        // Create block commit
+        let block_header = BlockHeader {
+            author: author.clone(),
+            prev_block_finalization_proof: finalization_proof.clone(),
+            previous_hash: Commit::Block(last_header.clone()).to_hash256(),
```
Comment (junha1) [inline repository/src/lib.rs:502]: `last_header.to_hash256()`

File: repository/src/lib.rs [hunk 5]
```diff
@@ -424,9 +425,112 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+                "branch {} should be rebased on {}",
+                WORK_BRANCH_NAME,
+                FINALIZED_BRANCH_NAME
+            ));
+        }
+
+        // Check the validity of the commit sequence
+        let commits = read_commits(self, last_header_commit, work_commit).await?;
+        let last_header = self.get_last_finalized_block_header().await?;
+        let mut reserved_state = self.get_reserved_state().await?;
```
Comment (junha1) [inline repository/src/lib.rs:450]: Please checkout to the target branch (`work` in this case) first before retrieving the RS.

File: repository/src/lib.rs (hunk 1 above)
Comment (TomTaehoonKim) [reply repository/src/lib.rs:520]: https://github.com/postech-dao/simperby/blob/main/repository/src/lib.rs#L456

I was keeping it in the same manner.. I'm not sure what to do here..

File: repository/src/lib.rs (hunk 2 above)
Comment (TomTaehoonKim) [reply repository/src/lib.rs:478]: fixed

File: repository/src/lib.rs (hunk 4 above)
Comment (TomTaehoonKim) [reply repository/src/lib.rs:490]: the value is the same right?

File: repository/src/lib.rs (hunk 3 above)
Comment (TomTaehoonKim) [reply repository/src/lib.rs]: fixed

File: repository/src/lib.rs (hunk 5 above)
Comment (TomTaehoonKim) [reply repository/src/lib.rs]: understood and fixed

File: repository/src/lib.rs (hunk 4 above)
Comment (junha1) [reply repository/src/lib.rs:584]: yeap

Comment (TomTaehoonKim) [reply repository/src/lib.rs]: fixed

File: repository/src/lib.rs (hunk 1 above)
Comment (junha1) [reply repository/src/lib.rs:612]: I think this is not that important.

**Review by junha1 — APPROVED** [review]

**Pull request merged**